### ISSUE TITLE
Incorrect parsing of the default log format when using parse_nginx_log

### ISSUE THREAD

**Issue by f1rstC0de** [vectordotdev/vector#13325]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

The type string is processed incorrectly:
`172.30.156.184 - i.imangulov [24/Jun/2022:18:32:02 +0300] "GET /v1/statement/queued/20220624_153201_00009_wjqc8/y3160e5a3b3f5e23c3764463a7433dbe6ecd9efee/2 HTTP/1.1" 200 335 "-" "Trino JDBC Driver/351"`

With a line like:
`127.0.0.1 - - [24/Jun/2022:18:29:49 +0300] "GET /stub_status HTTP/1.1" 200 105 "-" "NGINX-Prometheus-Exporter/v0.10.0"`

Or:
`172.30.133.21 - - [24/Jun/2022:09:46:53 +0300] "GET /ui/api/query HTTP/1.1" 200 7420 "https://trino.agu.com/ui/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.5 Safari/605.1.15"`

There are no problems.
The problem definitely lies in the remote_user field.

### Configuration

```text
[sources.logs_{{ item.key }}]
  type    = "file"
  include = {{ item.value.path }}

[transforms.parse_logs_{{ item.key }}]
  inputs = ["logs_{{ item.key }}"]
  type = "remap"
  source = '''
    . = parse_nginx_log!(.message,"combined")
    '''

[transforms.add_fields_logs_{{ item.key }}]
  inputs = ["parse_logs_{{ item.key }}"]
  type = "remap"
  source = """
  .inst = get_hostname!()
  .env = "{{ vector_sage_env }}"
  .group = "{{ vector_sage_group }}"
  .system = "{{ vector_sage_system }}"
  .type = "nginx_log"
  """

[transforms.drop_fields_logs_{{ item.key }}]
  inputs = ["add_fields_logs_{{ item.key }}"]
  type = "remap"
  source = """
  ."@timestamp" = del(.timestamp)
  del(.file)
  del(.host)
  del(.source_type)
  """

[sinks.out_{{ item.key }}]
  type = "kafka"
  inputs = ["drop_fields_logs_{{ item.key }}"]
  compression = "gzip"
  bootstrap_servers = {{ vector_kafka_hosts | to_json }}
  key_field = "{{ vector_sage_group }}"
  topic = "sage-logs-{{ vector_sage_group }}"
  librdkafka_options."message.max.bytes" = "{{ vector_max_message_bytes }}"
  librdkafka_options."client.id" = "{{ vector_sage_group }}_logs"
  encoding = "json"
```

### Version

vector 0.22.2 (x86_64-unknown-linux-gnu 0024c92 2022-06-15)

### Example Data

`172.30.156.184 - i.imangulov [24/Jun/2022:18:32:02 +0300] "GET /v1/statement/queued/20220624_153201_00009_wjqc8/y3160e5a3b3f5e23c3764463a7433dbe6ecd9efee/2 HTTP/1.1" 200 335 "-" "Trino JDBC Driver/351"`

### Additional Context

nginx version: nginx/1.20.1
Linux 3.10.0-1160.53.1.el7.x86_64 #1 SMP Fri Jan 14 13:59:45 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Nginx config:
```
user nginx;
worker_processes auto;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;


events {
        worker_connections 768;
}

http {

        sendfile on;
        tcp_nopush on;
        tcp_nodelay on;
        keepalive_timeout 65;
        types_hash_max_size 2048;
        include /etc/nginx/mime.types;
        default_type application/octet-stream;

        ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3; # Dropping SSLv3, ref: POODLE
        ssl_prefer_server_ciphers on;

        access_log /var/log/nginx/access.log;
        error_log /var/log/nginx/error.log;
        gzip on;
        include /etc/nginx/conf.d/*.conf;
        include /etc/nginx/sites-enabled/*;
}
```
That is, the standard logging format is used. 
Same as described [here](https://serverfault.com/questions/916653/what-is-the-default-log-format-on-nginx)

**Comment by jszwedko** [comment]
Thanks @f1rstC0de . It does seem like `remote_user` is the issue here.

**Comment by f1rstC0de** [comment]
It seems to me that some characters are not processed, such as `.`
Maybe a regular expression will help `[\w\d_-\./]+` ?
I think this should cover 99.9% of possible `remote_user` names.
However, a large number of cases need to be tested. Very non-standard naming options can come across)

**Comment by f1rstC0de** [comment]
Hello! I see that the fix was not included in release `0.22.3`. Will it enter `0.22.4`? Or will it be easier and faster to rewrite the pipeline using `parse_regex` for the time being?

**Comment by jszwedko** [comment]
Hi @f1rstC0de ! This fix will be included in 0.23.0 which is targeted for next week.

**Comment by f1rstC0de** [comment]
🔥  Thanks!
